lib/ramble/ramble/application.py

```diff
@@ -2216,88 +2221,111 @@ def _analysis_dicts(self, criteria_list):
+        if self.workflow_manager is not None:
+            fom_sources.append(self.workflow_manager)
+
+        all_contexts = {}
+        for source in fom_sources:
+            for when_fs, source_context_defs in source.figure_of_merit_contexts.items():
+                if self.expander.satisfies(list(when_fs), variant_set=self.object_variants):
+                    for context, context_def in source_context_defs.items():
+                        all_contexts[context] = context_def
+            extra_vars = (
+                source.modded_variables(self) if source.origin_type == "modifier" else None
+            )
+            # figures_of_merit[frozenset(when_list)][frozenset(context_list)][fom_name]
+            for when_fs, source_contexts in source.figures_of_merit.items():
+                if not self.expander.satisfies(list(when_fs), variant_set=self.object_variants):
+                    continue
 
-            mod_vars = mod.modded_variables(self)
+                for context_fs, source_foms in source_contexts.items():
+                    if not context_fs:  # FOMs with no defined context are set to the null context
+                        context_fs = frozenset([_NULL_CONTEXT])
+                        all_contexts[_NULL_CONTEXT] = {}
+                    for context in context_fs:
+                        if context not in all_contexts:
+                            fom_list = str(list(source_foms.keys()))
+                            logger.die(
+                                f"Figure(s) of merit {fom_list} registered to context "
+                                f"'{context}', which is not found. Check FOM and FOM context "
+                                "definitions and 'when' conditions."
+                            )
 
-            for fom, fom_def in mod.figures_of_merit.items():
-                fom_definitions[fom] = {"origin": f"{mod}", "origin_type": "modifier"}
-                for attr in fom_def.keys():
-                    if isinstance(fom_def[attr], (list, FomType)):
-                        fom_definitions[fom][attr] = fom_def[attr].copy()
-                    else:
-                        fom_definitions[fom][attr] = self.expander.expand_var(
-                            fom_def[attr], mod_vars
-                        )
+                        # Copy context definition for contexts used by a FOM
+                        if context not in definitions:
+                            definitions[context] = {
+                                "definition": {},
+                                "foms": {},
+                            }
+                            if context != _NULL_CONTEXT:
+                                regex_str = self.expander.expand_var(
+                                    all_contexts[context]["regex"]
+                                )
+                                definitions[context]["definition"] = {
+                                    "regex": re.compile(r"%s" % regex_str),
```

Comment:
Can we replace `r"%s" % regex_str` with just `regex_str` now? (Since we're editing this code anyway)

@linsword13 is there any reason this shouldn't work?